lib/frontend.js

```diff
@@ -107,67 +112,26 @@ function SASL(type, x, options) {
 }
 
 async function SASLContinue(type, x, options, pass) {
-  if (!hasCryptoSupport)
-    throw errors.generic({
-      message: 'Auth type ' + (authNames[type] || type) + ' require a working Crypto API',
-      type: authNames[type] || type,
-      code: 'AUTH_TYPE_NOT_IMPLEMENTED'
-    })
   const res = toString(x, 9).split(',').reduce((acc, x) => (acc[x[0]] = x.slice(2), acc), {})
 
-  const saltedPassword = await crypto.subtle.generateKey(
-    {
-      name: 'PBKDF2',
-      salt: encodeBase64(res.s),
-      iterations: parseInt(res.i, 10),
-      length: 32,
-      hash: {
-        name: 'SHA-256'
-      }
-    },
-    false,
-    ['deriveKey', 'deriveBits']
-  )
-
-  const clientKey = await crypto.subtle.importKey(
-    'raw',
-    fromString('Client Key'),
-    {
-      name: 'HMAC',
-      hash: { name: 'SHA-256' }
-    },
-    true,
-    ["sign", "verify"]
-  )
-  const exportedClientKey = await crypto.subtle.exportKey(
-    'raw',
-    clientKey
-  )
+  const saltedPassword = await pbkdf2Derive(
```

Comment:
The first parameter of `pbkdf2Derive` and `hmacSign` is always `'sha256'`, it be refactored to removed the first parameter I guess